test/state/state.cpp

```diff
@@ -90,6 +90,96 @@ TransactionCost compute_tx_intrinsic_cost(evmc_revision rev, const Transaction&
     return {intrinsic_cost, min_cost};
 }
 
+int64_t process_authorization_list(
+    State& state, uint64_t chain_id, const AuthorizationList& authorization_list)
+{
+    int64_t delegation_refund = 0;
+    for (const auto& auth : authorization_list)
+    {
+        // 1. Verify the chain id is either 0 or the chain’s current ID.
+        if (auth.chain_id != 0 && auth.chain_id != chain_id)
+            continue;
+
+        // 2. Verify the nonce is less than 2**64 - 1.
+        if (auth.nonce == Account::NonceMax)
+            continue;
+
+        // 3. Verify if the signer has been successfully recovered from the signature.
+        //    authority = ecrecover(...)
+        if (!auth.signer.has_value())
+            continue;
+
+        // s value must be less than or equal to secp256k1n/2, as specified in EIP-2.
+        if (auth.s > SECP256K1N_OVER_2)
+            continue;
+
+        // Get or create the authority account.
+        // It is still empty at this point until nonce bump following successful authorization.
+        auto& authority = state.get_or_insert(*auth.signer, {.erase_if_empty = true});
+
+        // 4. Add authority to accessed_addresses (as defined in EIP-2929.)
+        authority.access_status = EVMC_ACCESS_WARM;
+
+        // 5. Verify the code of authority is either empty or already delegated.
+        if (authority.code_hash != Account::EMPTY_CODE_HASH &&
+            !is_code_delegated(state.get_code(*auth.signer)))
+            continue;
+
+        // 6. Verify the nonce of authority is equal to nonce.
+        // In case authority does not exist in the trie, verify that nonce is equal to 0.
+        if (auth.nonce != authority.nonce)
+            continue;
+
+        // 7. Add PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST gas to the global refund counter
+        // if authority exists in the trie.
+        // Successful authorisation validation makes an account non-empty.
+        // We apply the refund only if the account has existed before.
+        // We detect "exists in the trie" by inspecting _empty_ property (EIP-161) because _empty_
+        // implies an account doesn't exist in the state (EIP-7523).
+        //
+        // For completeness, we also consider _empty_ accounts in the state as existent
+        // and apply refund for those. The erase_if_empty indicates the account has been created
+        // as empty while processing this authorization list.
+        // TODO: Such accounts are against EIP-7523 and may happen only in testing environments.
+        const auto empty_in_state = authority.is_empty() && !authority.erase_if_empty;
```

Comment:
Maybe `in_state_but_empty` ?

Reply:
I'd also mention that it means that storage is non-empty

Reply:
> I'd also mention that it means that storage is non-empty

This is not accurate. There can be even fully empty account.